### PR TITLE
Updates WordPress to 5.0.1

### PR DESCRIPTION
**Pull request by matt-bernhardt**
## Status
_Use labels (`review needed`, `in progress`, or `paused`) to declare status_

#### What does this PR do?
This will update WordPress to 5.0.1 (eventually)

#### How can a reviewer manually see the effects of these changes?
Pantheon will comment in this thread with a link to the review app. Log into that app (see Matt for the credentials) and check the version of WordPress.

#### Requires new or updated plugins, themes, or libraries?
NO

#### Requires change to deploy process?
NO

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   "require": {
     "composer/installers": "^1.3.0",
     "pantheon-systems/quicksilver-pushback": "^1.0",
-    "pantheon-systems/wordpress-composer": "^4.8.2",
+    "pantheon-systems/wordpress-composer": "^5.0.0",
```
Comment (frrrances) [inline composer.json:20]: looks like it's still at 5.0.0?

Comment (matt-bernhardt) [reply composer.json:20]: The way Composer requirements get expressed is that this change means "at least 5.0.0, until you get to 6.0.0" - so we wouldn't need to update `composer.json` with any further changes until WP6 comes out.

Comment (matt-bernhardt) [reply composer.json:20]: Composer's treatment of versions and the `^` or `~` or other characters is documented here, btw - I'm working on internalizing this today: https://getcomposer.org/doc/articles/versions.md

**Review by frrrances — APPROVED** [review]
Makes sense!  :shipit:

**Pull request merged**